Sources/PackageGraph/PubGrub/PubGrubDependencyResolver.swift

```diff
@@ -514,9 +514,9 @@ public struct PubGrubDependencyResolver {
             return .conflict
         }
 
-        self.delegate?.derived(term: unsatisfiedTerm.inverse)
-        state.derive(unsatisfiedTerm.inverse, cause: incompatibility)
 
+        state.derive(unsatisfiedTerm.inverse, cause: incompatibility)
+        self.delegate?.derived(term: unsatisfiedTerm.inverse)
```

Comment:
Is it meaningful that the order changed here?

Reply:
no, it was part of a change I reverted (I made state.derive throwing so wanted to call delegate only if succeeded), but should not have impact. the delegate is used for logging etc